core/heap.h

```diff
@@ -281,7 +281,11 @@ void
 nonpersistent_heap_free(dcontext_t *dcontext, void *p,
                         size_t size HEAPACCT(which_heap_t which));
 
-/* Passing dcontext == GLOBAL_DCONTEXT allocates from a global pool. */
+/* Passing dcontext == GLOBAL_DCONTEXT allocates from a global pool.
+ * Important note: within the W^X scheme, this will return an address of the writeable
```

Comment:
I would name the option in parens or somewhere: -satisfy_w_xor_x.

Reply:
Done.